### PR TITLE
added issues data model

### PR DESCRIPTION
**Pull request by KarlPatach**

### CODE REVIEW

File: force-app/main/default/authproviders/Github.authprovider-meta.xml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <AuthProvider xmlns="http://soap.sforce.com/2006/04/metadata">
     <authorizeUrl>https://github.com/login/oauth/authorize</authorizeUrl>
-    <consumerKey>Iv1.ffbd85c3bd59fcb6</consumerKey>
-    <consumerSecret>SYdwqyoJGybpZJi9wlpxXPnpHk6avtXz5sjuIVUCTatPZG+KOaITUnBBGPdiC2rX</consumerSecret>
+    <consumerKey>Iv1.e44c321504db8640</consumerKey>
```
Comment (juanka588) [inline force-app/main/default/authproviders/Github.authprovider-meta.xml:4]: sorry I replaced your credentials :P

File: force-app/main/default/classes/IssueDataModel.cls [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        issue.put('RepositoryURL', issue.get('repository_url'));
+
+        issue.put('Number', issue.get('number'));
+
+        issue.put('State', issue.get('state'));
+        issue.put('Title', issue.get('title'));
+        issue.put('Body', issue.get('body'));
+
+
+        issue.put('CreatedDate', issue.get('created_at'));
```
Comment (juanka588) [inline force-app/main/default/classes/IssueDataModel.cls:62]: use DataUtils.transformDate((String)instance)

File: force-app/main/default/classes/IssueDataModel.cls [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        Map<String,Object> pullRequest=(Map<String,Object>)issue.get('pull_request');
+        if(pullRequest!=null && pullRequest.containsKey('url')){
+          issue.put('PullRequest', pullRequest.get('url'));
+        }
+        
+        return issue;
+    } 
+
+    public static DataSource.TableResult handleQuery(DataSource.QueryContext queryContext){
```
Comment (juanka588) [inline force-app/main/default/classes/IssueDataModel.cls:85]: remove this

File: force-app/main/default/classes/IssueDataModel.cls [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        //TODO
+       return handleQuery(queryContext,'apache','lucene-solr');
+    } 
+    
+    public static DataSource.TableResult handleQuery(DataSource.QueryContext queryContext, String user, String repositoryName){
```
Comment (juanka588) [inline force-app/main/default/classes/IssueDataModel.cls:90]: remove unused params

File: force-app/main/default/classes/IssueDataModel.cls (hunk 3 above)
Comment (juanka588) [reply force-app/main/default/classes/IssueDataModel.cls:85]: please

File: force-app/main/default/classes/IssueDataModel.cls (hunk 2 above)
Comment (juanka588) [reply force-app/main/default/classes/IssueDataModel.cls:62]: and BTW check for null inside the method please

File: force-app/main/default/authproviders/Github.authprovider-meta.xml (hunk 1 above)
Comment (KarlPatach) [reply force-app/main/default/authproviders/Github.authprovider-meta.xml:4]: Np, I guess this file should be .gitignored for future commits

File: force-app/main/default/classes/IssueDataModel.cls (hunk 2 above)
Comment (KarlPatach) [reply force-app/main/default/classes/IssueDataModel.cls:62]: Check for null for the instance ? in DataUtils.transformDate((String)instance)

Comment (juanka588) [reply force-app/main/default/classes/IssueDataModel.cls:62]: inside the method in that case return ''

File: force-app/main/default/classes/DataUtils.cls [hunk 5]
```diff
@@ -0,0 +1,8 @@
+public class DataUtils {
+    // Hacky but quick way to get a string date in a compatible DateTime format.
+    // To avoid EXTERNAL_OBJECT_EXCEPTION in SOQL when requesting a date field
+    // '2020-06-19T01:45:17Z' -> '2020-06-19 01:45:17'
+    public static String transformDate(String rawDate){
+      return rawDate.replace('T', ' ').replace('Z', '');
```
Comment (juanka588) [inline force-app/main/default/classes/DataUtils.cls:6]: put the null check here

File: force-app/main/default/classes/IssueDataModel.cls [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        issue.put('RepositoryURL', issue.get('repository_url'));
+
+        issue.put('Number', issue.get('number'));
+
+        issue.put('State', issue.get('state'));
+        issue.put('Title', issue.get('title'));
+        issue.put('Body', issue.get('body'));
+
+
+        issue.put('CreatedDate', issue.get('created_at')==null?null:DataUtils.transformDate((String)issue.get('created_at')));
```
Comment (juanka588) [inline force-app/main/default/classes/IssueDataModel.cls:62]: add the cast only

**Pull request merged**